gulpfile.js

```diff
@@ -66,9 +89,110 @@ gulp.task('ext:compile-src', (done) => {
         .pipe(srcmap.write('.', {
             sourceRoot: function(file){ return file.cwd + '/src'; }
         }))
-        .pipe(gulp.dest('out/src/'));
+        .pipe(gulp.dest('out/src/views/htmlcontent'));
+});
+
+// Copy systemjs config
+gulp.task('ext:copy-systemjs-config', (done) => {
+    return gulp.src([
+        config.paths.project.root + '/src/views/htmlcontent/*.js'])
+        .pipe(gulp.dest('out/src/views/htmlcontent'));
+});
+
+// Copy css
+gulp.task('ext:copy-css', (done) => {
+    return gulp.src([
+            config.paths.project.root + '/src/views/htmlcontent/src/css/*.css'])
+        .pipe(gulp.dest('out/src/views/htmlcontent/src/css'));
+});
+
+// Copy images
+gulp.task('ext:copy-images', (done) => {
+    return gulp.src([
+            config.paths.project.root + '/src/views/htmlcontent/src/images/**/*'])
+        .pipe(gulp.dest('out/src/views/htmlcontent/src/images'));
+});
+
+// Clean angular slickgrid library
+gulp.task('ext:clean-angular2-slickgrid', function() {
+    return del(config.paths.project.root + '/node_modules/angular2-slickgrid/**/*.ts');
+});
```

Comment:
Is there a reason you wouldn't just have this delete the entire directory completely?

Reply:
yes, we need the compiled js files

Reply:
Normally clean would be run before a full fresh build so it'd be expected that you'd be pulling down the node_modules again anyways - why are you just deleting the ts files? What is the purpose of doing that?

---
In reply to: [321910002](https://github.com/microsoft/vscode-mssql/pull/1282#discussion_r321910002) [](ancestors = 321910002)

Reply:
because it causes compilation issues and typescript projects aren't supposed to have .ts files in the compiled part

Reply:
That's fine - I have no issues with deleting the files. I'm just saying that "clean" has a very specific meaning and that this isn't really doing a clean as I would consider it. I'd at least suggest updating the name to be something like remove-angular2-slickgrid-sourcefiles